File: py2many/bubble.py
def bubble_sort(size):
    a = [random.randrange(0, 10000) for i in range(size)]

    for i in range(size - 1, 0, -1):
        for j in range(0, i):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]

    print(a)

File: py2many/bubble.go
package main

import (
"fmt"
iter "github.com/hgfischer/go-iter")



func BubbleSort[T0 any](size T0 any) {
a := iter.NewIntSeq(iter.Start(0), iter.Stop(size)).All().iter().map(|i| randrange(random, 0, 10000)).collect::<Vec<_>>()
for _, i := range iter.NewIntSeq(iter.Start((size - 1)), iter.Stop(0), iter.Step(-1)).All() {
for _, j := range iter.NewIntSeq(iter.Start(0), iter.Stop(i)).All() {
if(a[j] > a[(j + 1)]) {
{
var __tmp1, __tmp2 = a[(j + 1)], a[j]
a[j] = __tmp1
a[(j + 1)] = __tmp2
}
}
}
}
fmt.Printf("%v\n",a);}


